templates/src/objective-c/BoringSSL-GRPC.podspec.template

```diff
@@ -225,25 +223,14 @@
       % endfor
       EOF
 
-      # The symbol prefixing mechanism is performed by redefining BoringSSL symbols with "#define
-      # SOME_BORINGSSL_SYMBOL GRPC_SHADOW_SOME_BORINGSSL_SYMBOL". Unfortunately, some symbols are
-      # already redefined as macros in BoringSSL headers in the form "#define SOME_BORINGSSL_SYMBOL
-      # SOME_BORINGSSL_SYMBOL" Such type of redefinition will cause "SOME_BORINGSSL_SYMBOL redefined"
-      # error when using together with our prefix header. So the workaround in the below lines removes
-      # all such type of #define directives.
-      sed -i'.back' '/^#define \\([A-Za-z0-9_]*\\) \\1/d' include/openssl/*.h
-      # Remove lines of the format below for the same reason above
-      #     #define SOME_BORINGSSL_SYMBOL ${"\\"}
-      #         SOME_BORINGSSL_SYMBOL
-      sed -i'.back' '/^#define.*\\\\$/{N;/^#define \\([A-Za-z0-9_]*\\) *\\\\\\n *\\1/d;}' include/openssl/*.h
+      # Grab prefix header from Github repo
+      curl -o include/openssl/boringssl_prefix_symbols.h -L https://raw.githubusercontent.com/grpc/grpc/master/src/objective-c/boringssl_prefix_headers/boringssl_prefix_symbols-${boringssl_commit}.h
 
       # We are renaming openssl to openssl_grpc so that there is no conflict with openssl if it exists
       find . -type f \\( -path '*.h' -or -path '*.cc' -or -path '*.c' \\) -print0 | xargs -0 -L1 sed -E -i'.grpc_back' 's;#include <openssl/;#include <openssl_grpc/;g'
-    END_OF_COMMAND
+      # BoringSSL include boringssl_prefix_symbols.h without any prefix, which does not match the
 
-    # Redefine symbols to avoid conflict when the same app also depends on OpenSSL. The list of
-    # symbols are src/objective-c/grpc_shadow_boringssl_symbol_list.
-    # This is the last part of this file.
-    s.prefix_header_contents = 
-      ${expand_symbol_list(settings.grpc_shadow_boringssl_symbols)}
+      # Xcode import style. We add it here so that Xcode knows where to find it.
+      find . -type f \\( -path '*.h' -or -path '*.cc' -or -path '*.c' \\) -print0 | xargs -0 -L1 sed -E -i'.grpc_back' 's;#include <boringssl_prefix_symbols.h>;#include <openssl_grpc/boringssl_prefix_symbols.h>;g'
```

Comment:
seems dirty, can we avoid this line?

Reply:
I don't think so because if users build their projects as Framework, Apple does require Framework name to be the first segment in the include statement (see [Apple's doc](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Tasks/IncludingFrameworks.html)). The only possible way to avoid it is to hack the "Header search path" of the generated `gRPC-Core` and `BoringSSL-GRPC` targets pointing to the downloaded BoringSSL repo. Not sure if it'll work for Framework (because in that case the two projects are including a header outside of the Framework's Header directory). And if it works, IMHO it's an even more dirty hack.